src/shogun/features/StringFeatures.cpp

```diff
@@ -78,19 +78,22 @@ template<class ST> CStringFeatures<ST>::CStringFeatures(CAlphabet* alpha)
 }
 
 template<class ST> CStringFeatures<ST>::CStringFeatures(const CStringFeatures & orig)
-: CFeatures(orig), num_vectors(orig.num_vectors),
-	single_string(orig.single_string),
-	length_of_single_string(orig.length_of_single_string),
-	max_string_length(orig.max_string_length),
-	num_symbols(orig.num_symbols),
-	original_num_symbols(orig.original_num_symbols),
-	order(orig.order), preprocess_on_get(false),
-	feature_cache(NULL)
+: CFeatures(orig)
 {
 	init();
 
 	ASSERT(orig.single_string == NULL) //not implemented
 
+	num_vectors=orig.num_vectors;
+	single_string=orig.single_string;
```

Comment:
so should this be `single_string=std::string(orig.single_string)` to get a copy?

Reply:
See this reply https://github.com/shogun-toolbox/shogun/pull/4423#discussion_r249283822

So if we need shallow copy then this is fine. However, my understanding is that in the future we will use std::string instead of char* for strings? Is that correct @karlnapf ?

Reply:
as far as i can remember std::string is mapped in every swig interface, so std::string should be preferred yes

Reply:
I pushed one more change. I think we should not change the data type for this PR since it is just for fixing this particular issue.

Reply:
agreed. Changing strings is a whole big other story